src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6337,7 +6543,88 @@ private IOperation VisitNoneOperationExpression(IOperation operation)
 
         public override IOperation? VisitInterpolatedStringHandlerArgumentPlaceholder(IInterpolatedStringHandlerArgumentPlaceholderOperation operation, int? captureIdForResult)
         {
-            return VisitNoneOperation(operation, captureIdForResult);
+            AssertContainingContextIsForThisCreation(operation, _currentInterpolatedStringHandlerContext);
+            switch (operation.PlaceholderKind)
+            {
+                case InterpolatedStringArgumentPlaceholderKind.TrailingValidityArgument:
+                    if (_currentInterpolatedStringHandlerContext.OutPlaceholder == -1)
+                    {
+                        throw ExceptionUtilities.Unreachable;
+                    }
+
+                    return new FlowCaptureReferenceOperation(_currentInterpolatedStringHandlerContext.OutPlaceholder, operation.Syntax, operation.Type, operation.GetConstantValue(), isInitialization: true);
+
+                case InterpolatedStringArgumentPlaceholderKind.CallsiteReceiver:
+                    if (_currentInterpolatedStringHandlerContext.HasReceiver && tryGetArgumentOrReceiver(-1) is IOperation receiverCapture)
+                    {
+                        Debug.Assert(receiverCapture is IFlowCaptureReferenceOperation);
+                        return OperationCloner.CloneOperation(receiverCapture);
+                    }
+                    else
+                    {
+                        return new InvalidOperation(ImmutableArray<IOperation>.Empty, semanticModel: null, operation.Syntax, operation.Type, operation.GetConstantValue(), isImplicit: true);
+                    }
+
+                case InterpolatedStringArgumentPlaceholderKind.CallsiteArgument:
+                    if (tryGetArgumentOrReceiver(operation.ArgumentIndex) is IOperation argumentCapture)
+                    {
+                        Debug.Assert(argumentCapture is IFlowCaptureReferenceOperation or IDiscardOperation);
+                        return OperationCloner.CloneOperation(argumentCapture);
+                    }
+                    else
+                    {
+                        return new InvalidOperation(ImmutableArray<IOperation>.Empty, semanticModel: null, operation.Syntax, operation.Type, operation.GetConstantValue(), isImplicit: true);
+                    }
+
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(operation.PlaceholderKind);
+            }
+
+            IOperation? tryGetArgumentOrReceiver(int argumentIndex)
+            {
+                int startingStackDepth = _currentInterpolatedStringHandlerContext.StartingStackDepth;
+                if (startingStackDepth >= _evalStack.Count)
```

Comment:
>if (startingStackDepth >= _evalStack.Count) [](http://example.com/codeflow?start=16&length=43)

It looks like we are not handling the ```-1``` case, which is apparently can happen in an error recovery scenario.

Reply:
I don't believe this case is currently possible.